sim/src/make/generator.rs

```diff
@@ -418,29 +418,41 @@ fn rand_time(rng: &mut XorShiftRng, low: Time, high: Time) -> Time {
 impl ScenarioGenerator {
     // Designed in https://github.com/dabreegster/abstreet/issues/154
     pub fn proletariat_robot(map: &Map, rng: &mut XorShiftRng, timer: &mut Timer) -> Scenario {
-        // First classify buildings into residences (with a number of people) or workplaces. No
-        // mixed-use yet; every building is one or the other.
         let mut residences: Vec<(BuildingID, usize)> = Vec::new();
         let mut workplaces: Vec<BuildingID> = Vec::new();
         let mut total_ppl = 0;
         for b in map.all_buildings() {
-            // These are scraped from OSM "shop" and "amenity" tags.
-            if b.amenities.is_empty() {
-                // TODO Guess number of residences based on OSM tags.
-                let num_ppl = rng.gen_range(1, 5);
-                residences.push((b.id, num_ppl));
-                total_ppl += num_ppl;
-            } else {
-                workplaces.push(b.id);
+            match b.bldg_type {
+                BuildingType::Residential(num_ppl) => {
+                    residences.push((b.id, num_ppl));
+                    total_ppl += num_ppl;
+                }
+                BuildingType::ResidentialCommercial(num_ppl) => {
+                    residences.push((b.id, num_ppl));
+                    total_ppl += num_ppl;
+                    workplaces.push(b.id);
+                }
+                BuildingType::Commercial => {
+                    workplaces.push(b.id);
+                }
+                BuildingType::Empty => {}
             }
         }
 
         let mut s = Scenario::empty(map, "random people going to/from work");
+        s.only_seed_buses = None;
         timer.start_iter("create people", total_ppl);
         for (home, num_ppl) in residences {
             for _ in 0..num_ppl {
                 timer.next();
                 // Make a person going from their home to a random workplace, then back again later.
+                
+                // TODO refactor
+                // function or associated item not found in `rand_xorshift::XorShiftRng`
+                // so why it works in buildings.rs?
+                //let mut deterministicRng = XorShiftRng::seed_from_u64(home.0 as u64);
+                //let work = *workplaces.choose(deterministicRng).unwrap();
```

Comment:
So maps itself should be stable, but model has no such requirements as it is anyway regenerated?

I though about preserving the same trips between making changes (lanes/parking lanes modifications and so on). Or is it guaranteed already anyway?

Reply:
> So maps itself should be stable, but model has no such requirements as it is anyway regenerated?

Correct. If you run `importer` multiple times using the same code and .osm, you should get the same results; that's why we have to seed the RNG using OSM IDs. This scenario isn't generated at import-time; it's created last minute when the player chooses it. (This keeps release size small and avoids us having to re-run the importer when we change this algorithm.) The RNG passed in here happens to be fixed right now across game sessions and all time (it has a hardcoded seed). This is mainly done so that when the player is working on some of the main challenge modes, they can compare their results to a baseline and see an exact change. In the future, maybe this RNG can vary with the player's control; they could say "generate a slight variation of this traffic pattern."

> Or is it guaranteed already anyway?

That should already be taken care of. The player can't make any edits today that would affect this scenario; the building types are all completely fixed. Player edits can affect simulation setup, so there's stuff like https://github.com/dabreegster/abstreet/blob/6b337c10ec0101b55ed9f65bde629bbcb9571607/sim/src/make/scenario.rs#L265 to remain stable. If you added/removed a single parking lane and suddenly the initial position of many other parked cars far away changed, that would be confusing. There's a few other uses of `fork_rng` like that. But I don't think it's a concern here.

Reply:
Ok, now I understand this. Thank you!